articles/build-rules.md

```diff
@@ -0,0 +1,89 @@
+---
+authorName: Vassilis Rizopoulos
+authorGithubUsername: damphyr
+---
+# TBD
+
+This article discusses the application of the [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) concept in Build & CI systems. 
+
+It is layed out as a set of rules to follow when creating build and CI infrastructure for your project. 
+
+Think of each rule as a [T.A.R.D.I.S.](https://en.wikipedia.org/wiki/TARDIS): they are bigger on the inside. 
+
+## Distinctions, distinctions
+
+To identify the separate concerns we should provide definitions for the terms: 
+
+A **build system** performs transformations in sequence according to a predetermined dependency chain to create artifacts. A subset of this is the compilation of sources to binaries.
+
+A **build management system** coordinates build system(s).
+
+### Specificity
+
+A build system is highly project specific. It is affected by toolchains, project conventions and generally can only run on a specially configured host (i.e. development environment).
+
+A build management system can run everywhere as long as it can start the build system on the appropriate host. The specificity of a build management system is limited to the number of version control systems it supports. Although theoretically it doesn't have to provide version control support it is a given that such support will be provided in the minimum feature set.
+
+### The Makefile is the build system
+
+Make, rake, Ant, MSBuild, graddle, grunt etc. **are not build systems**<sup>1</sup>. You create build systems with them. 
+
+Jenkins, TeamCity, TravisCI, BuildBot, TFS etc. are build management systems.
+
+## The Rules
+
+### 1. I am the build server
+
+Rule #1 requires that the build server follows the exact same steps as any other developer. 
+
+Expressed the other way around: Every developer has to be able to recreate locally the complete build process without deviations given the development environment and the correct version of the source tree.
+
+### 2. When the build server says no, it means no!
+
+Rule #2 says that if a build server marks a build as broken, then the build is broken. Drop everything and read the logs. 
+
+There is no "it works for me", your build server is Judge Dredd: judge, juror and executioner.
```

Comment:
Again a popcultural reference I did not know. I am learning new stuff here on all levels!

Reply:
You cannot be that young! ;)

Reply:
There is also a third sci-fi culture reference in the article. lets see who spots it.